extra/3-magic-8-ball.js

```diff
@@ -45,23 +47,73 @@
 
 // This should log "The ball has shaken!"
 // and return the answer.
+
+  let arr = [
+    "It is certain.",
+    "It is decidedly so.",
+    "Without a doubt.",
+    "Yes - definitely.",
+    "You may rely on it.",
+
+    "As I see it, yes.",
+    "Most likely.",
+    "Outlook good.",
+    "Yes.",
+    "Signs point to yes.",
+
+    "Reply hazy, try again.",
+    " Ask again later.",
+    "Better not tell you now.",
+    "Cannot predict now.",
+    "Concentrate and ask again.",
+
+    "Don't count on it.",
+    "My reply is no.",
+    "My sources say no.",
+    "Outlook not so good.",
+    "Very doubtful.",
+  ];
+
 function shakeBall() {
-  //Write your code in here
+  console.log("The ball has shaken!");
+  return arr[Math.floor(Math.random() * 20)];
+
 }
 
+
 /* 
   This function should say whether the answer it is given is
     - very positive
-    - positive
+    - positive 
     - negative
     - very negative
 
   This function should expect to be called with any value which was returned by the shakeBall function.
 */
 function checkAnswer(answer) {
-  //Write your code in here
+  // find the index of the answer
+    // use a loop to go through the arr
+  // based on the index, we'll know which group the answer is in
+  for(let i = 0; i <= arr.length -1  ; i++){
```

Comment:
great for loop. Another way to check this would be to use the `.indexOf()` method on the array, i.e.
```
let index = arr.indexOf(answer);
if (index < 5){
...
```